RadixWallet/Clients/AccountPortfoliosClient/AccountPortfoliosClient+State.swift

```diff
@@ -0,0 +1,285 @@
+		guard var portfolio = portfoliosSubject.value.wrappedValue?[address] else {
+			return
+		}
+		portfolio.stakeUnitDetails = stakeUnitDetails.map { details in
+			var details = details
+			details.updateFiatWorth(calculateWorth(gateway))
+			return details
+		}
+		setOrUpdateAccountPortfolio(portfolio)
+	}
+}
+
+// MARK: Fiat Worth changes
+private extension AccountPortfoliosClient.AccountPortfolio {
+	mutating func updateFiatWorth<T>(value: T, to keyPath: WritableKeyPath<FiatWorth, T>) {
+		updateFiatWorth { _, worth in
+			var worth = worth.fiatWorth
+			worth?[keyPath: keyPath] = value
+			return worth
+		}
+	}
+
+	mutating func updateFiatWorth(_ change: (ResourceAddress, ResourceAmount) -> FiatWorth?) {
+		account.fungibleResources.updateFiatWorth(change)
+		stakeUnitDetails.mutateValue { $0.updateFiatWorth(change) }
+		poolUnitDetails.mutateValue { $0.updateFiatWorth(change) }
+	}
+}
+
+private extension OnLedgerEntity.OwnedFungibleResources {
+	mutating func updateFiatWorth(_ change: (ResourceAddress, ResourceAmount) -> FiatWorth?) {
+		xrdResource.mutate { resource in
+			resource.amount.fiatWorth = change(.mainnetXRDAddress, resource.amount)
+		}
+
+		nonXrdResources.mutateAll { resource in
+			resource.amount.fiatWorth = change(resource.resourceAddress, resource.amount)
+		}
+
+		nonXrdResources.sort(by: <)
+	}
+}
+
+private extension MutableCollection where Element == OnLedgerEntitiesClient.OwnedResourcePoolDetails {
+	mutating func updateFiatWorth(_ change: (ResourceAddress, ResourceAmount) -> FiatWorth?) {
+		mutateAll { detail in
+			detail.xrdResource?.redemptionValue.mutate { amount in
+				amount.fiatWorth = change(.mainnetXRDAddress, amount)
+			}
+
+			detail.nonXrdResources.mutateAll { resource in
+				let address = resource.resource.resourceAddress
+				resource.redemptionValue.mutate { amount in
+					amount.fiatWorth = change(address, amount)
+				}
+			}
+		}
+	}
+}
+
+private extension MutableCollection where Element == OnLedgerEntitiesClient.OwnedStakeDetails {
+	mutating func updateFiatWorth(_ change: (ResourceAddress, ResourceAmount) -> FiatWorth?) {
+		mutateAll { detail in
+			detail.stakeUnitResource.mutate {
+				$0.amount.fiatWorth = change(.mainnetXRDAddress, $0.amount)
+			}
+			detail.stakeClaimTokens.mutate {
+				$0.stakeClaims.mutateAll { token in
+					token.claimAmount.fiatWorth = change(.mainnetXRDAddress, token.claimAmount)
+				}
+			}
+		}
+	}
+}
+
+// MARK: - Account portfolio fiat worth
+extension AccountPortfoliosClient.AccountPortfolio {
+	var totalFiatWorth: Loadable<FiatWorth> {
+		poolUnitDetails.concat(stakeUnitDetails).map { poolUnitDetails, stakeUnitDetails in
+			let totalFiatWorth = account.fungibleResources.fiatWorth + stakeUnitDetails.fiatWorth + poolUnitDetails.fiatWorth
+			return .init(isVisible: isCurrencyAmountVisible, worth: totalFiatWorth, currency: fiatCurrency)
+		}
+		.errorFallback(.unknownWorth(isVisible: isCurrencyAmountVisible, currency: fiatCurrency))
```

Comment:
if loading poolUnit or stakeUnits failed, fallback to unknown worth, that is displayed differently in the UI.